File: com.codemachine/others/Aditya Verma/Binary Search/OnceAppearingElement.java
class OnceAppearingElement{
    static int onceAppearingElement(int[] Array, int low, int high){
        if(low > high){
            return -1;
        }
        int mid = low + (high - low) / 2;
        int prev = (mid + Array.length - 1) % Array.length;
        int next = (mid + 1) % Array.length;
        if (Array[mid] != Array[prev] && Array[mid] != Array[next])
            return Array[mid];
        else if ((Array[mid] == Array[next] && mid % 2 == 0) || (Array[mid] == Array[prev] && mid % 2 != 0))
            return onceAppearingElement(Array, mid+1, high);
        else
            return onceAppearingElement(Array, low, mid-1);
    }
    public static void main(String[] args){
        int[] Array = {1,1,2,2,3,3,4,4,5};
        int data = onceAppearingElement(Array, 0, Array.length-1);
        System.out.println(data);
    }
}

File: Coder File/Binary Search/OnceAppearingElement.py
def OnceAppearingElement(lst):
    low = 0
    high = len(lst)-1
    while low <= high:
        mid = low+(high-low)//2
        previousElement = (mid+len(lst)-1) % len(lst)
        NextElement = (mid+1) % len(lst)
        if lst[mid] != lst[previousElement] and lst[mid] != lst[NextElement]:
            return lst[mid]
        elif (mid % 2 == 0 and lst[mid] == lst[NextElement]) or (mid % 2 != 0 and lst[mid] == lst[previousElement]):
            low = NextElement
        else:
            high = previousElement
    return -1

lst = [1,1,2,2,3,3,4,4,5,5,6,6,7,7,8]
data = OnceAppearingElement(lst)
if data == -1:
    print("Array contains Twice appearing elements")
else:
    print(f"Once appearing element is : {data}")